vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go

```diff
@@ -169,8 +169,15 @@ const (
 	// UpdateModeAuto means that autoscaler assigns resources on pod creation
 	// and additionally can update them during the lifetime of the pod,
 	// using any available update method. Currently this is equivalent to
-	// Recreate, which is the only available update method.
+	// Recreate.
 	UpdateModeAuto UpdateMode = "Auto"
+	// UpdateModeInPlaceOrRecreate means that autoscaler tries to assign resources in-place.
+	// If this is not possible (e.g., resizing takes too long or is infeasible), it falls back to the
+	// "Recreate" update mode.
+	// Requires VPA level feature gate "InPlaceOrRecreate" to be enabled
+	// on the admission and updater pods.
```

Comment:
In the context of the whole project, this makes sense.
In the context of this single PR landing into master, this doesn't make sense (since those components haven't been updated yet).
I assume we're fine with this inconsistency in master for a while, for a work in progress feature?

Reply:
We are merging into the feature branch https://github.com/kubernetes/autoscaler/tree/in-place-updates, to avoid this, if that makes sense :-)

Then when everything is merged in, I'll PR the feature-branch into main for a final review.

Reply:
🤦 I didn't realise that, my bad.